test/e2e/parallel/spacerequest_test.go

```diff
@@ -0,0 +1,172 @@
+				// given
+				// something/someone updates the tierName directly on the Space object
+
+				// when
+				subSpace, err = hostAwait.UpdateSpace(t, subSpace.Name, func(s *toolchainv1alpha1.Space) {
+					s.Spec.TierName = "base" // let's change the tier
+				})
+				require.NoError(t, err)
+
+				// then
+				// spaceRequest should reset back the tierName
+				_, err = awaitilities.Host().WaitForSpace(t, subSpace.GetName(),
+					UntilSpaceHasTier("appstudio"), // tierName is back as the one on spaceRequest
+					UntilSpaceHasAnyTargetClusterSet(),
+					UntilSpaceHasAnyProvisionedNamespaces(),
+				)
+				require.NoError(t, err)
+
+				t.Run("delete space request", func(t *testing.T) {
+					// now, delete the SpaceRequest and expect that the Space will be deleted as well,
+					// along with its associated namespace
+
+					// when
+					err := memberAwait.Client.Delete(context.TODO(), spaceRequest)
+
+					// then
+					// subSpace should be deleted as well
+					require.NoError(t, err)
+					err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(t, subSpace.Name)
+					require.NoError(t, err)
+					err = memberAwait.WaitUntilNSTemplateSetDeleted(t, subSpace.Name)
+					require.NoError(t, err)
+					err = memberAwait.WaitUntilNamespaceDeleted(t, subSpace.Name, "appstudio")
+					require.NoError(t, err)
+				})
+			})
+		})
+	})
+}
+
+func TestUpdateSpaceRequest(t *testing.T) {
+	// given
+	t.Parallel()
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	// when
+	spaceRequest, subSpace := CreateSpaceRequest(t, awaitilities, memberAwait.ClusterName,
+		WithSpecTierName("appstudio"),
+		WithSpecTargetClusterRoles([]string{cluster.RoleLabel(cluster.Tenant)}))
+
+	// then
+	spaceRequestNamespacedName := types.NamespacedName{Namespace: spaceRequest.Namespace, Name: spaceRequest.Name}
+	_, err := memberAwait.WaitForSpaceRequest(t, spaceRequestNamespacedName,
+		UntilSpaceRequestHasTierName("appstudio"),
+		UntilSpaceRequestHasConditions(Provisioned()),
+	)
+	require.NoError(t, err)
+
+	VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name,
+		UntilSpaceHasAnyTargetClusterSet(),
+		UntilSpaceHasTier(spaceRequest.Spec.TierName),
+	)
+
+	t.Run("update space request tierName", func(t *testing.T) {
+		// when
+		_, err := memberAwait.UpdateSpaceRequest(t, spaceRequestNamespacedName,
+			func(s *toolchainv1alpha1.SpaceRequest) {
+				s.Spec.TierName = "base"
+			},
+		)
+		require.NoError(t, err)
+
+		//then
+		// wait for both spaceRequest and subSpace to have same tierName
+		_, err = memberAwait.WaitForSpaceRequest(t, spaceRequestNamespacedName,
+			UntilSpaceRequestHasTierName("base"),
+			UntilSpaceRequestHasConditions(Provisioned()),
+		)
+		require.NoError(t, err)
+		_, err = hostAwait.WaitForSpace(t, subSpace.Name,
+			UntilSpaceHasTier("base"),
+			UntilSpaceHasConditions(Provisioned()))
```

Comment:
Good idea, added ! Please see [32f2a7c](https://github.com/codeready-toolchain/toolchain-e2e/pull/685/commits/32f2a7cc9eb5347d00c19dbbd3a44a54ecbcfb4c)

Reply:
I saw it passing multiple times with this new `TerminatingSpace` condition, but apparently it may introduce some instability , since the transition from `Ready=false` to `Ready=true` could happen very quickly and the test may not be able to see it. For example :
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/codeready-toolchain_toolchain-e2e/685/pull-ci-codeready-toolchain-toolchain-e2e-master-e2e/1638544948812845056/build-log.txt

WDYT?

Reply:
in this particular case, it could be also because you rely on the "Reason=Terminating" - if you would just simply check the `Status=false` and don't care about the reason then it could be fine. 
But you are right about the concern that it could cause flaky tests. That was just an idea, but didn't realize the consequence - feel free to drop it :+1:

Reply:
Ok, then let's drop it for now and see if it's more stable.

Dropped in [de6a70c](https://github.com/codeready-toolchain/toolchain-e2e/pull/685/commits/de6a70c71b6d5c1f78a987f397dba4a8354da23b)